sdk/eventhub/event-hubs/src/diagnostics/messageSpan.ts

```diff
@@ -0,0 +1,31 @@
+import { SpanContext, Span, TracerProxy, SpanKind, getTraceParent } from '@azure/core-tracing';
+import { EventData } from '../eventData';
+
+export const TRACEPARENT_PROPERTY = "Diagnostic_Id";
+
+export function instrumentEventData(eventData: EventData, span: Span): boolean {
+  if (eventData.properties && eventData.properties[TRACEPARENT_PROPERTY]) {
+    return false;
+  }
+
+  try {
+    const traceParent = getTraceParent(span.context());
```

Comment:
I wonder if we should consider a way of parsing that doesn't throw, like returning null or some other sentinel value, since exceptions-as-control-flow is a little more expensive.

Reply:
I agree, and think null or undefined would be good choices. I think instead of throwing an error, the `getTraceParent` method could log instead (I don't think logging has been added to core-tracing yet but it should eventually).

Reply:
Anders would say to use `undefined`